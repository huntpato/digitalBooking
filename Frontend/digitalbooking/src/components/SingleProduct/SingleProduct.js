import React, { Fragment, useEffect, useState, useMemo } from "react";
import { useLocation } from "react-router-dom";
import { ProductScore } from "../ProductScore/ProductScore";
import { ProductBanner} from "../ProductBanner/ProductBanner"
import useFetch from "use-http";
import { GridBanner } from "../GridBanner/GridBanner";
import { NotFound } from '../NotFound/NotFound'
import DatePickerProduct from "../DatePickerProduct/DatePickerProduct";
import { BookingCard } from "../BookingCard/BookingCard";
import styles from "./SingleProduct.module.css";
import { Maps } from "../Maps/Maps";
import { ProductService } from "../ProductService/ProductService";
import { GalleryContext } from "../ProductService/GalleryContext";
import { Policies } from "../Policies/Policies"
import ImagesCarousel from "../ImagesCarousel/ImagesCarousel";

export const SingleProduct = () => {
  const location = useLocation();
  const path = location.pathname.split("/");
  const id = path[2];
  const options = {headers:{
    "X-Requested-With": "XMLHttpRequest"
  }};
  const url = `https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/products/${id}`;
  const { loading, data = [] } = useFetch(url, options, []);
  const {product,info,locDetail,description,dates,datesBlock,map, mapInfo, calendar, servicios, infoContainer, galleryContainer,galleryEventContainer} = styles;
  
  const [gallery, setGallery] = useState(false)
  const value = useMemo(() => ({ gallery, setGallery }), [gallery, setGallery ]);

  useEffect(() => {
    window.scrollTo({
      top: 0,
    });
  }, []);


  return !loading ? (
    <Fragment class= "singleProd">
      {data.productName &&
      <div className={product}>
        <ProductBanner title={ data.productCategory.name } subtitle={data.productName} />
        <section className={info}>
          <div className={infoContainer}>
            <div>
              <div className={locDetail}>
                <i className="fas fa-map-marker-alt"></i>
                <h4>{data.city.name}</h4>
              </div>
              <p> A 950 metros del centro</p>
            </div>
            <ProductScore rate={data.review} />
          </div>
        </section>
        <GalleryContext.Provider value= { value } >
        {gallery && <div className={galleryEventContainer}><ImagesCarousel imagesGallery={data.photoGallery}/></div>}
        <GridBanner
          ph1={data.photoGallery[0]}
          ph2={data.photoGallery[1]}
          ph3={data.photoGallery[2]}
          ph4={data.photoGallery[3]}
          ph5={data.photoGallery[4]}
          ph6={data.photoGallery[5]}
          productId ={id}
        />
        </GalleryContext.Provider>
        <div className={galleryContainer}>
          <ImagesCarousel imagesGallery={data.photoGallery}/>
        </div>
        <div className={description}>
          <h4>Alojate en {data.location}</h4>
          <p>{data.productDescription}</p>
        </div>
        <section className={servicios}>
          <h4>¿Qué ofrece este lugar?</h4>
          <hr/>
          <ProductService productServices={data.services}/>
        </section>
        <section className={dates}>
          <div className={datesBlock}>
            <h4>Fechas Disponibles</h4>
            <div className={calendar}>
              <DatePickerProduct />
              <BookingCard path= {data.id}/>
            </div>
          </div>
        </section>
        <section className={map}>
          <div className={mapInfo}>
            <h4>¿Dónde vas a estar?</h4>
            <hr/>
            <h5> {data.city.name} </h5>
          </div>
          <Maps
            lat={data.coordinates[0]}
            long={data.coordinates[1]}
            name={data.productName}
            city={data.location}
            width= "94%"
            height="80vh"
            margin= "3%"
          />
        </section>
        <Policies hrules={data.houseRules} health= {data.healthSafety} cancel ={data.cancellationPolicy}/>
      </div>} 
      {!data.productName && <NotFound/>}
    </Fragment>
  ) : (
    <div>Loading...</div>
  );
};
