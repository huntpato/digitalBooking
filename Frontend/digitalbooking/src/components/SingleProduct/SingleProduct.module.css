.product {
  display: flex;
  flex-direction: column;
  min-height: 120vh;
  padding: 68px 0 0 0;
  width: 100%;
  -webkit-animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
  animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
}

.product p {
  font-size: 14px;
}

/* title product */

.header h5 {
  padding-bottom: 4px;
}

.header i {
  color: white;
}

/* location title section */

.info {
  align-items: center;
  background-color: var(--background-color2);
  /*background-color: coral;*/
  display: flex;
  flex-direction: row;
  justify-content: center;
  padding: 10px 0px 10px 0px;
  width: 100%;
}

.infoContainer{
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: baseline;
  width: 90vw;
}

.info p {
  font-size: 14px;
  padding: 5px 0 0 18px;
}

.locDetail {
  align-self: center;
  display: flex;
  flex: row;
  padding-left: 0px;
  width: 100%;
}

.locDetail i {
  color: var(--color-typo2);
  padding-top: 1px;
  padding-right: 5px;
  margin:0;
}

.locationDetail {
  align-self: flex-start;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  padding-bottom: 10px;
}

/*description section*/

.description {
  align-self: center;
  display: flex;
  flex-direction: column;
  padding: 0px 0px 0px 0px;
  width: 90vw;
}

.description h4{
  align-self: flex-start;
  font-size: 18px;
}

.description p {
  align-self: flex-start;
  padding-top: 10px;
  margin-bottom: 40px;
  
}

/*services section*/

.servicios{
  align-self: center;
  display:flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 90vw;
}

.servicios h4{
  margin-bottom: 10px;
  font-size: 18px;
}

.servicios hr{
  background-color:var(--principal-color);
  border: none;
  width: 100%;
  height: 1.4px;
  margin-bottom: 40px;
}


/*calendar section*/

.dates{
    background-color: var(--background-color);
    width: 100%;
    height: 100%;
    display:flex;
    flex-direction: column;
    align-items: center;
    padding-bottom: 40px;
}

.dates h4{
    font-size: 18px;
    margin: 10px 0px 40px;
    align-self: flex-start;
    
}

.datesBlock{
    display:flex;
    flex-direction: column;
    align-content: center;
    margin-top: 30px;
    /*height: 300px;*/
    width: 90vw;
}

.calendar{
  display:flex;
  flex-direction: row;
  justify-content: center; 
}

/*map section*/

.map{
  align-self: center;
  display:flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-top: 30px;
  width: 90vw;
}

.mapInfo{
  width:100%;
}

.mapInfo h4{
  font-size: 18px;
  margin: 20px 0px 10px 0px;
  align-self: flex-start;
}

.mapInfo hr{
  background-color:var(--principal-color);
  border: none;
  width: 100%;
  height: 1.4px;
  margin-bottom: 10px;
}


/*mobile gallery*/
.galleryContainer{
  visibility:hidden;
}

/*desktop gallery*/
.galleryEventContainer{
  height: 93vh;
  width:100%;
  position:fixed;
  z-index:10;
  padding-top: 0px;
  background-color: rgba(56, 59, 88, 0.9);
}


@media (max-width: 1046px){

  /*calendar section */

  .dates{
      
    padding-bottom: 50px; 
  }

  .dates h4{
    
    margin: 10px 0px 35px;
    align-self: flex-start;
    
  }

  .calendar{
    
    display:flex;
    flex-direction: column;
    align-items: center;
    
  }
}

@media (max-width: 768px) {

  .product {
    min-height: 140vh;
  }

  .galleryEventContainer{
    display:none;
  }

  /*mobile gallery*/
  .galleryContainer{
    visibility:visible;
  }

  .description {
    padding: 0px 0 0px 0px;
  }

  /* calendar section */

  .dates{
    padding-bottom: 50px; 
  }

  .dates h4{
    margin: 10px 0px 15px;     
  }

}

@media (max-width: 480px) {

  .product {
    min-height: 140vh;
    justify-content: center;
  }

  .info p {
    display: none;
  }

  .locDetail {
    width: 155px;
  }

  .locDetail h4 {
    font-size: 14px;
  }

  /* calendar rules */

  .dates{
      
    padding-bottom: 30px; 
  }

}

@-webkit-keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
@keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

