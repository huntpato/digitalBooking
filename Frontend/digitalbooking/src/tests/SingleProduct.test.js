import React from 'react';
import { shallow } from 'enzyme';
import { SingleProduct } from '../components/SingleProduct/SingleProduct'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';


jest.mock('react-router-dom', () => ({
  useParams: jest.fn().mockReturnValue({ nifUuid: 'nif123' }),
  useHistory: jest.fn(),
  useLocation:jest.fn().mockReturnValue({
    pathname: "/register",
    search: "",
    hash: "",
    state: null,
    key: "5nvxpbdafa",
  }),
}));

describe('Pruebas en componente <SingleProduct />', () => {
    const wrapper= shallow(<SingleProduct/>);
   
    test('Comprobar que <SingleProduct /> renderiza correctamente.', () => {
      
    expect( wrapper ).toMatchSnapshot();
        
     
    })

    
})
