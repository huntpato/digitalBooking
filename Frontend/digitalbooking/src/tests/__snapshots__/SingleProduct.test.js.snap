// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <SingleProduct /> Comprobar que <SingleProduct /> renderiza correctamente. 1`] = `
<div>
  Loading...
</div>
`;
